test/io.test.ts

```diff
@@ -149,21 +149,11 @@ it('modifies incoming server event', async () => {
       connection.server.send(event.data)
     })
 
-    io.server.on('greeting', (message) => {
+    io.server.on('greeting', function (message) {
```

Comment:
Downside: server listeners must not be arrow functions (will rebind them) in order to access the original context. Not an issue if you aren't planning on preventing the default.